### PR TITLE
add logit, expit and log_expit to cupyx.scipy.special

### PR DESCRIPTION
**Pull request by grlee77**
This PR adds three very simple statistical functions to `cupyx.scipy.special`. This PR is built on top of #5687 to avoid conflicts with the changes to `__init__.py` made there. Only the last commit here is new.

cc @IvanYashchuk

### CODE REVIEW

**Comment by grlee77** [comment]
SciPy [defines these functions](https://github.com/scipy/scipy/blob/d5497bdc3c1a5ce1432c8ca6f1ab92c87937f74d/scipy/special/functions.json#L567-L572) for single ('f'), double ('d') and long double ('g') dtypes while the implementation here is for single or double only.

**Comment by grlee77** [comment]
rebased after #5687 was merged. should be much easier to review now

File: cupyx/scipy/special/_statistics.py [hunk 1]
```diff
@@ -1,6 +1,102 @@
...
+    preamble=log_expit_definition,
+    doc='''Logarithm of the logistic sigmoid function.
+
+    Args:
+        x (cupy.ndarray): input data (must be real)
+
+    Returns:
+        cupy.ndarray: values of log(expit(x))
+
+    .. seealso:: :data:`scipy.special.expit`
```
Comment (kmaehashi) [inline cupyx/scipy/special/_statistics.py:91]:
```suggestion
    .. seealso:: :data:`scipy.special.log_expit`
```

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**